### ISSUE TITLE
New install on Ububtu 20.04 - manager and api containers keep crashing

### ISSUE THREAD

**Issue by nandonov** [kasmtech/workspaces-issues#275]
Hi there,
First of all, this seems to be an issue on Ubuntu 20.04 and 22.04 as I first tried to install KASM on Ubuntu 22.04 (not my preferred platform but I do see why you guys are fleeing all Red Hat distro, I would have done the same with all the changes around them lately) and I had the same issue.

Also, I would like to point out that I got through all open and closed issues to see if there is something similar to what I have stumbled on and there is nothing I could find close to my issue.

Here you have my setup:

nandonov@kasmatic:~$ df -h
Filesystem                         Size  Used Avail Use% Mounted on
udev                               1.9G     0  1.9G   0% /dev
tmpfs                              394M  1.4M  392M   1% /run
/dev/mapper/ubuntu--vg-ubuntu--lv   97G   19G   74G  20% /
tmpfs                              2.0G     0  2.0G   0% /dev/shm
tmpfs                              5.0M     0  5.0M   0% /run/lock
tmpfs                              2.0G     0  2.0G   0% /sys/fs/cgroup
/dev/loop1                          33M   33M     0 100% /snap/snapd/12704
/dev/loop0                          56M   56M     0 100% /snap/core18/2128
/dev/loop2                          71M   71M     0 100% /snap/lxd/21029
/dev/sda2                          974M  106M  801M  12% /boot
tmpfs                              394M     0  394M   0% /run/user/1000

nandonov@kasmatic:~$ uname -a
Linux kasmatic 5.4.0-128-generic #144-Ubuntu SMP Tue Sep 20 11:00:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

 nandonov@kasmatic:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

nandonov@kasmatic:~$ free -h
              total        used        free      shared  buff/cache   available
Mem:          3.8Gi       324Mi       2.7Gi        14Mi       820Mi       3.3Gi
Swap:         7.8Gi          0B       7.8Gi

nandonov@kasmatic:~$ sudo docker system info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.9.1-docker)
  compose: Docker Compose (Docker Inc., v2.5.0)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 7
  Running: 5
  Paused: 0
  Stopped: 2
 Images: 8
 Server Version: 20.10.18
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-128-generic
 Operating System: Ubuntu 20.04.3 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.84GiB
 Name: kasmatic
 ID: AZAF:J4EU:QULW:Z55U:RBVF:IQXA:FLGT:RUPY:KLC7:W3UI:Y6MF:K5EU
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support

Now, here are the logs from the two containers which are crashing:

MANAGER

Executing /usr/bin/kasm_manager.so
Received config /opt/kasm/current/conf/app/api.app.config.yaml
cheroot/__init__.py:13: UserWarning: Module cheroot was already imported from /tmp/_MEILXZ7aP/cheroot/__init__.pyc, but /tmp/_MEILXZ7aP/cheroot-8.6.0-py3.8.egg is being added to sys.path
cheroot/__init__.py:13: UserWarning: Module more_itertools was already imported from /tmp/_MEILXZ7aP/more_itertools/__init__.pyc, but /tmp/_MEILXZ7aP/more_itertools-8.12.0-py3.8.egg is being added to sys.path
cheroot/__init__.py:13: UserWarning: Module six was already imported from /tmp/_MEILXZ7aP/six.pyc, but /tmp/_MEILXZ7aP/six-1.16.0-py3.8.egg is being added to sys.path
cherrypy/__init__.py:112: UserWarning: Module cherrypy was already imported from /tmp/_MEILXZ7aP/cherrypy/__init__.pyc, but /tmp/_MEILXZ7aP/CherryPy-18.1.1-py3.8.egg is being added to sys.path
cherrypy/__init__.py:112: UserWarning: Module portend was already imported from /tmp/_MEILXZ7aP/portend.pyc, but /tmp/_MEILXZ7aP/portend-2.6-py3.8.egg is being added to sys.path
cherrypy/__init__.py:112: UserWarning: Module tempora was already imported from /tmp/_MEILXZ7aP/tempora/__init__.pyc, but /tmp/_MEILXZ7aP/tempora-5.0.1-py3.8.egg is being added to sys.path
cherrypy/__init__.py:112: UserWarning: Module pytz was already imported from /tmp/_MEILXZ7aP/pytz/__init__.pyc, but /tmp/_MEILXZ7aP/pytz-2022.1-py3.8.egg is being added to sys.path
2022-10-10 21:05:40,348 [INFO] manager_api_server: Performing Database Connectivity Test
Traceback (most recent call last):
  File "manager_api_server.py", line 916, in <module>
  File "manager_api_server.py", line 839, in start
  File "manager_api_server.py", line 809, in __init__
KeyError: 'manager'
[9] Failed to execute script 'manager_api_server' due to unhandled exception!

API

Executing /usr/bin/kasm_server.so
Received config /opt/kasm/current/conf/app/api.app.config.yaml
Using parameters --enable-admin-api --enable-client-api --enable-public-api
cheroot/__init__.py:13: UserWarning: Module cheroot was already imported from /tmp/_MEIxODSjN/cheroot/__init__.pyc, but /tmp/_MEIxODSjN/cheroot-8.6.0-py3.8.egg is being added to sys.path
cheroot/__init__.py:13: UserWarning: Module more_itertools was already imported from /tmp/_MEIxODSjN/more_itertools/__init__.pyc, but /tmp/_MEIxODSjN/more_itertools-8.12.0-py3.8.egg is being added to sys.path
cherrypy/__init__.py:112: UserWarning: Module cherrypy was already imported from /tmp/_MEIxODSjN/cherrypy/__init__.pyc, but /tmp/_MEIxODSjN/CherryPy-18.1.1-py3.8.egg is being added to sys.path
cherrypy/__init__.py:112: UserWarning: Module portend was already imported from /tmp/_MEIxODSjN/portend.pyc, but /tmp/_MEIxODSjN/portend-2.6-py3.8.egg is being added to sys.path
cherrypy/__init__.py:112: UserWarning: Module tempora was already imported from /tmp/_MEIxODSjN/tempora/__init__.pyc, but /tmp/_MEIxODSjN/tempora-5.0.1-py3.8.egg is being added to sys.path
2022-10-10 20:21:11,271 [INFO] root: Performing Database Connectivity Test
2022-10-10 20:21:11,562 [INFO] root: Added Log Handler
2022-10-10 20:21:14,459 [INFO] admin_api_server: AdminApi initialized
Traceback (most recent call last):
  File "server.py", line 283, in <module>
  File "admin_api.py", line 37, in __init__
  File "provider_manager.py", line 29, in __init__
AttributeError: 'NoneType' object has no attribute 'value'
[9] Failed to execute script 'server' due to unhandled exception!
Executing /usr/bin/kasm_server.so
Received config /opt/kasm/current/conf/app/api.app.config.yaml
Using parameters --enable-admin-api --enable-client-api --enable-public-api

I have tried to troubleshoot the issue but it seems like some code issue and I did not want to start digging into the code, unless not right away. So if someone can take a look and point me out to where to look it will be helpeful.

Thanks!

**Comment by nandonov** [comment]
I wanted to add that the version I am installing is 1.11.0 and I have disabled the firewall.

**Comment by j-travis** [comment]
The `AttributeError: 'NoneType' object has no attribute 'value'`  and `[9] Failed to execute script 'manager_api_server' due to unhandled exception!` seem to indicate there is a problem with the database and/or its initialization.

I recommend uninstalling and trying to install again but capture the output from the installer as there may be an error reported there that indicates the root problem.

Also, did you install docker and/or docker compose before installing workspaces, or did you let workspaces handle those installs

**Comment by nandonov** [comment]
Hi j-travis,
Thanks for the quick response.
I will reinstall and give you the full log. Regarding the question, yes, I have only installed the security upgrades and let the installation script take care of installing all dependencies. I did not install anything before running the script.

**Comment by nandonov** [comment]
Well, I have tried another installation and this time everything went without any issue. I guess there was something wrong on some dependent package and the issue was fixed between this weekend and today.

For the record, I did again installed new copy of Ubuntu 20.04.3, did an upgrade at the end, and proceeded with the installation. Very quickly I saw a portion I never saw before, like you said, the database was for some reason not initializing properly. Here is the section I saw for the first time today:

Dropping Existing Schema
Creating Schema
Creating Installation ID
Creating New Config Setting (logging:log_protocol) Value
Creating New Config Setting (logging:log_port) Value
Creating New Config Setting (logging:log_host) Value
Creating New Config Setting (logging:hec_token) Value
Creating New Config Setting (logging:https_insecure) Value
Creating New Config Setting (logging:http_method) Value
Creating New Config Setting (logging:url_endpoint) Value
Creating New Config Setting (logging:log_retention) Value
Creating New Config Setting (logging:debug_retention) Value
Creating New Config Setting (logging:minimize_local_logging) Value
Creating New Config Setting (auth:anonymous_user_expiration) Value
Creating New Config Setting (auth:notice_message) Value
Creating New Config Setting (auth:notice_title) Value
Creating New Config Setting (auth:login_assistance) Value
Creating New Config Setting (auth:enable_kasm_auth) Value
Creating New Config Setting (auth:kasm_auth_domain) Value
Creating New Config Setting (auth:max_login_attempts) Value
Creating New Config Setting (auth:same_site) Value
Creating New Config Setting (auth:session_lifetime) Value
Creating New Config Setting (cast:google_recaptcha_api_url) Value
Creating New Config Setting (cast:google_recaptcha_priv_key) Value
Creating New Config Setting (cast:google_recaptcha_site_key) Value
Creating New Config Setting (manager:update_check) Value
Creating New Config Setting (manager:agent_version) Value
Creating New Config Setting (manager:primary_manager_timeout) Value
Creating New Config Setting (manager:same_zone_reply) Value
Creating New Config Setting (manager:token) Value
Creating New Config Setting (manager:manager_expiration) Value
Creating New Config Setting (scale:guardian_interval) Value
Creating New Config Setting (scale:guardian_provision_threads) Value
Creating New Config Setting (scale:host_dead_expiration) Value
Creating New Config Setting (scale:host_missing_expiration) Value
Creating New Config Setting (scale:keepalive_expiration) Value
Creating New Config Setting (scale:provision_timeout) Value
Creating New Config Setting (scale:auto_agent) Value
Creating New Config Setting (images:add_images_to_default_group) Value
Creating New Config Setting (web_filter:web_filter_update_url) Value
Creating New Config Setting (licensing:license_server_url) Value
Creating Group (All Users)
Creating Group (Administrators)
Creating default group settings (administrator)
Creating default group settings (allow_kasm_audio)
Creating default group settings (allow_kasm_microphone)
Creating default group settings (allow_kasm_clipboard_down)
Creating default group settings (allow_kasm_clipboard_seamless)
Creating default group settings (allow_kasm_clipboard_up)
Creating default group settings (allow_kasm_downloads)
Creating default group settings (allow_kasm_uploads)
Creating default group settings (allow_persistent_profile)
Creating default group settings (allow_kasm_sharing)
Creating default group settings (allow_zone_selection)
Creating default group settings (display_ui_errors)
Creating default group settings (enable_ui_server_logging)
Creating default group settings (expose_user_environment_vars)
Creating default group settings (dashboard_redirect)
Creating default group settings (idle_disconnect)
Creating default group settings (inject_ssh_keys)
Creating default group settings (kasm_audio_default_on)
Creating default group settings (keepalive_expiration)
Creating default group settings (max_kasms_per_user)
Creating default group settings (enable_webp)
Creating default group settings (run_config)
Creating default group settings (default_image)
Creating default group settings (enable_totp_two_factor)
Creating default group settings (session_time_limit)
Creating default group settings (shared_session_full_control)
Creating default group settings (show_disabled_images)
Creating default group settings (require_subscription)
Creating default group settings (disabled_image_message)
Creating default group settings (kasm_ime_mode_default_on)
Creating default group settings (volume_mapping)
Creating default group settings (auto_login_to_kasm)
Creating default group settings (chat_history_messages)
Creating default group settings (lock_sharing_video_mode)
Creating default group settings (usage_limit)
Creating default group settings (web_filter_policy)
Creating default group settings (metadata)
Assigning Group Settings (allow_kasm_audio) to Group (All Users)
Assigning Group Settings (allow_kasm_microphone) to Group (All Users)
Assigning Group Settings (allow_kasm_clipboard_down) to Group (All Users)
Assigning Group Settings (allow_kasm_clipboard_seamless) to Group (All Users)
Assigning Group Settings (allow_kasm_clipboard_up) to Group (All Users)
Assigning Group Settings (allow_kasm_downloads) to Group (All Users)
Assigning Group Settings (allow_kasm_uploads) to Group (All Users)
Assigning Group Settings (allow_persistent_profile) to Group (All Users)
Assigning Group Settings (allow_kasm_sharing) to Group (All Users)
Assigning Group Settings (idle_disconnect) to Group (All Users)
Assigning Group Settings (kasm_audio_default_on) to Group (All Users)
Assigning Group Settings (keepalive_expiration) to Group (All Users)
Assigning Group Settings (max_kasms_per_user) to Group (All Users)
Assigning Group Settings (administrator) to Group (Administrators)
Assigning Group Settings (show_disabled_images) to Group (Administrators)
Assigning Group Settings (disabled_image_message) to Group (Administrators)
Creating User (admin@kasm.local)
Assigning User (admin@kasm.local) to Group (Administrators)
Creating User (user@kasm.local)
Creating Zone (default)

During the weekend only the lines below were there, did not had the schema creation and all the database setup, and the installer continued with the rest of the conatainers after that:

2022-10-12 01:47:51,952 [INFO] root: Performing Database Connectivity Test
2022-10-12 01:47:51,995 [INFO] root: Initializing Postgres Database
2022-10-12 01:48:28,540 [INFO] root: Stamping Database with revision
2022-10-12 01:48:40,707 [INFO] root: Complete

Unfortunately, I did not have any idea which package is different today, but the process I got through is the same, the fact that it did not work on both versions of ubuntu also confirms (somehow) that it was most probably some dependent package that caused the issue.

I will go ahead and close this now.
Thanks for your help.
